plugin-server/src/worker/plugins/run.ts

```diff
@@ -214,9 +215,11 @@ async function runSingleTeamPluginComposeWebhook(
 export async function runComposeWebhook(hub: Hub, event: PostIngestionEvent): Promise<void> {
     // Runs composeWebhook for all plugins for this team in parallel
     let pluginMethodsToRun = await getPluginMethodsForTeam(hub, event.teamId, 'composeWebhook')
-
     pluginMethodsToRun = await filterPluginMethodsForActionMatches(hub, event, pluginMethodsToRun)
 
+    // Here we load the special "LegacyActionWebhook" plugin which is actually run in process
+    pluginMethodsToRun = pluginMethodsToRun.concat(await getLegacyActionWebhookPluginsForTeam(hub, event))
+
     await Promise.all(
         pluginMethodsToRun.map(([pluginConfig, composeWebhook]) =>
             runSingleTeamPluginComposeWebhook(hub, event, pluginConfig, composeWebhook)
```

Comment:
I'd do something like this instead, i.e. here do fetching all the relevant pluginConfigs & then we can run the action matching in parallel with running composeWebhook (because it can be async due to cohorts). Action matching would happen within `getLegacyWebhook`
```suggestion
    // Get all PluginConfigs for legacy webhooks
    const legacyWebhookPluginConfigs = pluginConfigs.filter((pluginConfig) => {
                if (pluginConfig.plugin?.url !== '<docs-url>' || !pluginConfig.match_action_id) {
                    return
                }
    )
    
    await Promise.all(
        pluginMethodsToRun.map(([pluginConfig, composeWebhook]) =>
            runSingleTeamPluginComposeWebhook(hub, event, pluginConfig, composeWebhook)
        ),
        legacyWebhookPluginConfigs.map((pluginConfig) => runSingleTeamPluginComposeWebhook(hub, event, pluginConfig, (event) => getLegacyWebhook(pluginConfig, event))
   )
```